internal/k8s/reconciler/gateway.go

```diff
@@ -390,50 +390,7 @@ func (g *K8sGateway) ShouldBind(route store.Route) bool {
 }
 
 func (g *K8sGateway) Status() gwv1beta1.GatewayStatus {
-	listenerStatuses := []gwv1beta1.ListenerStatus{}
```

Comment:
The logic lives in `GatewayState` [here](https://github.com/hashicorp/consul-api-gateway/blob/eb308e0b33287eb7dbea7708cc659aaa24c2910c/internal/k8s/reconciler/state/gateway.go#L39-L69)